crates/net/network/src/session/active.rs

```diff
@@ -86,7 +86,8 @@ const MAXIMUM_TIMEOUT: Duration = INITIAL_REQUEST_TIMEOUT;
 /// How much the new measurements affect the current timeout (X percent)
 const SAMPLE_IMPACT: f64 = 0.1;
 /// Amount of RTTs before timeout
-const TIMEOUT_SCALING: u32 = 3;
+// This was updated from the initial value of 3, see https://github.com/paradigmxyz/reth/issues/1054s
+const TIMEOUT_SCALING: u32 = 10;
```

Comment:
this will have a greater impact for low rtt and max timeout is already clamped between min and max timeout.

I guess we can do this until we found a better solution with upcoming PRs

curios does this differ from geth's settings now?

Reply:
Geth's settings are defined here: https://github.dev/ethereum/go-ethereum/blob/55f41d198c133d672cdba258cdca40a9cb23005c/p2p/msgrate/msgrate.go

TIMEOUT_SCALING for geth is 3, but AFAIK they use it only to determine who to request the next header / block from for syncing, as the tracker is only created and used in the downloader subsystem. It doesn't result in kicking any peers AFAIK. They also keep track of a dynamic timeout per request type, which makes sense too.

Reply:
I see, so if a request times out, it just times out and there's no "punishment"?

I feel like we definitely need to treat synced/unsynced states differently and also tx related requests

Reply:
the main motivation for punishing timed out request was to maintain good peerset during sync, but as you highlighted this quickly becomes an issue when we start requesting transactions

Reply:
Yes, it says in the msgrate documentation that any systems can use this to free up peer slots, but it doesn't seem to be implemented anywhere. And yes I agree on that as well

Reply:
In their concurrent header downloader: [https://github.dev/ethereum/go-ethereum/blob/master/eth/downloader/fetchers_concurrent.go](https://github.dev/ethereum/go-ethereum/blob/master/eth/downloader/fetchers_concurrent.go#L156), there's this line:
```go
// timeoutGracePeriod is the amount of time to allow for a peer to deliver a
// response to a locally already timed out request. Timeouts are not penalized
// as a peer might be temporarily overloaded, however, they still must reply
// to each request. Failing to do so is considered a protocol violation.
var timeoutGracePeriod = 2 * time.Minute
```